pkg/controller/operators/catalog/operator.go

```diff
@@ -1067,9 +1077,40 @@ func (o *Operator) ExecutePlan(plan *v1alpha1.InstallPlan) error {
 						if err != nil {
 							return errorwrap.Wrapf(err, "error find matched CSV: %s", step.Resource.Name)
 						}
+						crd.SetResourceVersion(currentCRD.GetResourceVersion())
 						if len(matchedCSV) == 1 {
-							// Attempt to update CRD
-							crd.SetResourceVersion(currentCRD.GetResourceVersion())
+							o.logger.Debugf("Found one owner for CRD %v", crd)
```

Comment:
Don't block the PR on this, but I think there's some room for improving the readability here. Even something simple like replacing this block (~1082-1112) with a function `safeToUpdate(oldCrd, newCrd)` that we could test independently of the operator (and expand as we enable more methods).

Reply:
Followup - I'm a little concerned that there's no unit test for this (just an e2e) - so when you follow up with a unit test, refactoring like this might be an easy way to get that done quickly (because you can test a single, pure (w.r.t. fake clients) function)

Reply:
Added unit test and refactored to avoid using the clients and such.